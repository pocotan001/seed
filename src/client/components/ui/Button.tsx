import styled from "styled-components";
import { Color } from "../styles/enums";
import margin, { MarginProps } from "../styles/extends/margin";

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & MarginProps;

const Button = styled("button")<ButtonProps>`
  display: inline-flex;
  font-size: 1rem;
  line-height: 1.75;
  padding: 4px 16px;
  cursor: pointer;
  text-decoration: none;
  text-align: center;
  vertical-align: middle;
  color: ${Color.Grey800};
  border: 1px solid ${Color.Grey400};
  border-radius: 4px;
  background: transparent;
  align-items: center;
  justify-content: center;
  transition: background-color 300ms;

  &:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }

  &:disabled {
    cursor: default;
    opacity: 0.6;
  }

  ${margin};
`;

Button.defaultProps = {
  type: "button"
};

export default Button;
