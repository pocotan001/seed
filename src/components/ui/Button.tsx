import * as React from "react";
import styled, { css } from "styled-components";
import margin, { IMarginProps } from "~/components/styles/extends/margin";
import { colors } from "~/components/styles/theme";

interface IButtonProps
  extends IButtonStyleProps,
    React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
}

export interface IButtonStyleProps extends IMarginProps {
  block?: boolean;
}

const Button: React.SFC<IButtonProps> = ({
  children,
  block,
  m,
  mt,
  mr,
  mb,
  ml,
  ...rest
}) => <button {...rest}>{children}</button>;

Button.defaultProps = {
  type: "button"
};

export const buttonStyles = css<IButtonStyleProps>`
  font-size: 1rem;
  line-height: normal;
  display: ${({ block }) => (block ? "flex" : "inline-flex")};
  padding: 0.45em 0.75em;
  ${({ block }) => block && "width: 100%"};
  cursor: pointer;
  text-decoration: none;
  text-align: center;
  vertical-align: middle;
  color: ${colors.grey800};
  border: 1px solid ${colors.grey400};
  border-radius: 3px;
  background: ${colors.white};
  align-items: center;
  justify-content: center;

  &:hover {
    border-color: ${colors.grey500};
  }

  &:disabled {
    cursor: default;
    opacity: 0.6;
    border-color: ${colors.grey400};
  }

  ${margin};
`;

export default styled(Button)`
  ${buttonStyles};
`;
