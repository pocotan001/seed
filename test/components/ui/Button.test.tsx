import * as React from "react";
import * as renderer from "react-test-renderer";
import Button from "~/components/ui/Button";

describe("<Button>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Button>alo</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
