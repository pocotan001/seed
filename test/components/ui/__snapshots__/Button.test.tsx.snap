// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
<button
  className="sc-bdVaJa dEWpKO"
  type="button"
>
  alo
</button>
`;
